File: practice/implementation/chocolate_feast/chocolate_feast.py
#!/bin/python


t = int(raw_input().strip())
for _ in range(t):
    money, price, exch = map(int, raw_input().strip().split(' '))
    amount = money // price
    total = amount
    while amount >= exch:
        total += 1
        amount -= exch;
        amount += 1
    print total
    

File: practice/implementation/chocolate_feast/chocolate_feast.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

int main() 
{    
    int t, i = 0;
    int money, price, exch;
    cin >> t;
    while(i < t)
    {
        cin >> money >> price >> exch;
        int amount = money / price;
        int cnt = amount;
        while(amount >= exch)
        {
            cnt++;
            amount -= exch;
            amount++;
        }
        
        cout << cnt << endl;;
        i++;
    }
    
    return 0;
}

